src/createPersistor.js

```diff
@@ -53,15 +51,40 @@ export default function createPersistor (store, config) {
         }
 
         let key = storesToProcess.shift()
-        let storageKey = createStorageKey(key)
-        let endState = transforms.reduce((subState, transformer) => transformer.in(subState, key), stateGetter(store.getState(), key))
-        if (typeof endState !== 'undefined') storage.setItem(storageKey, serializer(endState), warnIfSetError(key))
+        persistCurrentStateForKey(state, key)
       }, debounce)
     }
 
     lastState = state
   })
 
+  function findStoresToProcess (state) {
+    let keys = []
+    stateIterator(state, (subState, key) => {
+      if (!passWhitelistBlacklist(key)) return
+      let newState = stateGetter(state, key)
+      if (stateGetter(lastState, key) === newState) return
+      keys.push(key)
+    })
+
+    return keys
+  }
+
+  function persistCurrentStateForKey (state, key) {
+    let storageKey = createStorageKey(key)
+    let currentState = stateGetter(state, key)
```

Comment:
I had issues with `let currentState = stateGetter(state, key)` getting a stale state.

This worked for me: `let currentState = stateGetter(store.getState(), key)`

I'm not sure if it was because of this or something else:

> The listener should not expect to see all state changes, as the state might have been updated multiple times during a nested dispatch() before the listener is called. It is, however, guaranteed that all subscribers registered before the dispatch() started will be called with the latest state by the time it exits.

http://redux.js.org/docs/api/Store.html#subscribe